### PR TITLE
provides a "close plugin library" button

### PR DESCRIPTION
**Pull request by tobibeer**
- starting from #1718 by @inmysocks
- possibly fixes all of #1718, #1597,  and #2068
- demo: http://2072.tiddlyspot.com

### CODE REVIEW

File: core/ui/ControlPanel/Modals/AddPlugins.tid [hunk 1]
```diff
@@ -5,7 +5,7 @@ subtitle: {{$:/core/images/download-button}} {{$:/language/ControlPanel/Plugins/
 
 \define install-plugin-button()
 <$button>
-<$action-sendmessage $message="tm-load-plugin-from-library" url={{!!url}} title={{$(assetInfo)$!!original-title}}/>
```
Comment (tobibeer) [inline core/ui/ControlPanel/Modals/AddPlugins.tid]: @Jermolene, appeared to be unused, correct?

Comment (Jermolene) [inline core/ui/ControlPanel/Modals/AddPlugins.tid]: No, the "title" parameter is needed to indicate which plugin should be loaded. I think the deletion in your pull request was a typo: the line below has a syntax error.

Comment (tobibeer) [inline core/ui/ControlPanel/Modals/AddPlugins.tid]: @Jermolene ...sorry, no idea what I was thinking ...now I see it being used "bright as daylight" ...reverted that change

File: core/modules/startup/browser-messaging.js [hunk 2]
```diff
@@ -93,6 +94,26 @@ exports.startup = function() {
 			});
 		}
 	});
+	// Listen for widget messages to control unloading the plugin library
+	$tw.rootWidget.addEventListener("tm-unload-plugin-library",function(event) {
+		var paramObject = event.paramObject || {},
+			url = paramObject.url;
+		$tw.browserMessaging.iframeInfoMap[url] = undefined;
+		if(url) {
```
Comment (Jermolene) [inline core/modules/startup/browser-messaging.js]: Please make `unloadIFrame()` into a separate helper function

File: core/modules/startup/browser-messaging.js [hunk 3]
```diff
@@ -93,6 +94,26 @@ exports.startup = function() {
...
+			$tw.utils.each(document.getElementsByTagName('iframe'), function(iframe) {
+				if(iframe.getAttribute("library") === "true" &&
+				  iframe.getAttribute("src") === url) {
+					iframe.parentNode.removeChild(iframe);
+				}
+			});
+			$tw.utils.each(
+				$tw.wiki.filterTiddlers("[[$:/temp/ServerConnection/" + url + "]] [prefix[$:/temp/RemoteAssetInfo/" + url + "/]]"),
```
Comment (Jermolene) [inline core/modules/startup/browser-messaging.js]: Perhaps worth adding `$tw.wiki.deleteTiddlers(filter)`.

File: core/modules/startup/browser-messaging.js (hunk 2 above)
Comment (tobibeer) [inline core/modules/startup/browser-messaging.js]: @Jermolene, the gain from that appears minimal, but ok

File: core/modules/startup/browser-messaging.js (hunk 3 above)
Comment (tobibeer) [inline core/modules/startup/browser-messaging.js]: @Jermolene, there appears to be no such function as of yet.

**Comment by tobibeer** [comment]
@Jermolene added **unloadIFrame** and minor syntax fixes

apparently, `$tw.wiki.deleteTiddlers(filter)` does not exist yet

**Pull request merged**